app/profile/[userId]/page.tsx

```diff
@@ -31,13 +36,43 @@ export default async function ProfilePage({
     },
   });
 
+  const { user: loggedInUser } = await validateRequest();
+
+  const loggedInUserId = loggedInUser?.id;
+  function isOwnProfile() {
+    return loggedInUser?.id === user?.id;
+  }
+
   if (!user) return <p>User not found</p>;
 
   const FRIENDS_SHOWN = 8;
 
+  async function logout(): Promise<ActionResult> {
```

Comment:
you can put that function in `actions/auth.ts`, then you dont have to pass it as a prop. You can just use it in the HeaderNav